includes/class-wc-payments.php

```diff
@@ -1389,4 +1400,20 @@ public static function enqueue_dev_runtime_scripts() {
 			wp_enqueue_script( 'WCPAY_RUNTIME', plugins_url( 'dist/runtime.js', WCPAY_PLUGIN_FILE ), [], self::get_file_version( 'dist/runtime.js' ), true );
 		}
 	}
+
+	/**
+	 * Display express checkout separator only when express buttons are displayed.
+	 *
+	 * @return void
+	 */
+	public static function display_express_checkout_separator_if_necessary() {
+		$woopay          = self::$platform_checkout_button_handler->should_show_platform_checkout_button() && self::$platform_checkout_button_handler->is_woopay_enabled();
+		$payment_request = self::$payment_request_button_handler->should_show_payment_request_button();
+		$should_hide     = $payment_request && ! $woopay;
```

Comment:
Out of curiosity - why add `display: none` instead of just short-circuiting and not outputting any HTML? (maybe a comment to explain this in the codebase would also be useful)

It seems that there could be a possibility of rendering multiple elements on the page with the same `id` of `wcpay-payment-request-button-separator`. Ideally, all elements that have an `id` on a single page should have a unique `id` attribute value.

Reply:
Discussed. We're good. The frontend is targeting the element, so that's why it's `display: none`.